// @rownd/react/dist/index.js
import D, { useEffect as Re, createContext as Ae, useRef as ye, useCallback as Y, useContext as Ie } from "react";
function $e(r) {
  return r && r.__esModule && Object.prototype.hasOwnProperty.call(r, "default") ? r.default : r;
}
var fe = { exports: {} }, N = { exports: {} }, O = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ve;
function Me() {
  if (ve)
    return O;
  ve = 1;
  var r = typeof Symbol == "function" && Symbol.for, w = r ? Symbol.for("react.element") : 60103, g = r ? Symbol.for("react.portal") : 60106, a = r ? Symbol.for("react.fragment") : 60107, f = r ? Symbol.for("react.strict_mode") : 60108, i = r ? Symbol.for("react.profiler") : 60114, m = r ? Symbol.for("react.provider") : 60109, s = r ? Symbol.for("react.context") : 60110, E = r ? Symbol.for("react.async_mode") : 60111, p = r ? Symbol.for("react.concurrent_mode") : 60111, c = r ? Symbol.for("react.forward_ref") : 60112, b = r ? Symbol.for("react.suspense") : 60113, A = r ? Symbol.for("react.suspense_list") : 60120, $ = r ? Symbol.for("react.memo") : 60115, j = r ? Symbol.for("react.lazy") : 60116, I = r ? Symbol.for("react.block") : 60121, M = r ? Symbol.for("react.fundamental") : 60117, k = r ? Symbol.for("react.responder") : 60118, u = r ? Symbol.for("react.scope") : 60119;
  function R(t) {
    if (typeof t == "object" && t !== null) {
      var L = t.$$typeof;
      switch (L) {
        case w:
          switch (t = t.type, t) {
            case E:
            case p:
            case a:
            case i:
            case f:
            case b:
              return t;
            default:
              switch (t = t && t.$$typeof, t) {
                case s:
                case c:
                case j:
                case $:
                case m:
                  return t;
                default:
                  return L;
              }
          }
        case g:
          return L;
      }
    }
  }
  function P(t) {
    return R(t) === p;
  }
  return O.AsyncMode = E, O.ConcurrentMode = p, O.ContextConsumer = s, O.ContextProvider = m, O.Element = w, O.ForwardRef = c, O.Fragment = a, O.Lazy = j, O.Memo = $, O.Portal = g, O.Profiler = i, O.StrictMode = f, O.Suspense = b, O.isAsyncMode = function(t) {
    return P(t) || R(t) === E;
  }, O.isConcurrentMode = P, O.isContextConsumer = function(t) {
    return R(t) === s;
  }, O.isContextProvider = function(t) {
    return R(t) === m;
  }, O.isElement = function(t) {
    return typeof t == "object" && t !== null && t.$$typeof === w;
  }, O.isForwardRef = function(t) {
    return R(t) === c;
  }, O.isFragment = function(t) {
    return R(t) === a;
  }, O.isLazy = function(t) {
    return R(t) === j;
  }, O.isMemo = function(t) {
    return R(t) === $;
  }, O.isPortal = function(t) {
    return R(t) === g;
  }, O.isProfiler = function(t) {
    return R(t) === i;
  }, O.isStrictMode = function(t) {
    return R(t) === f;
  }, O.isSuspense = function(t) {
    return R(t) === b;
  }, O.isValidElementType = function(t) {
    return typeof t == "string" || typeof t == "function" || t === a || t === p || t === i || t === f || t === b || t === A || typeof t == "object" && t !== null && (t.$$typeof === j || t.$$typeof === $ || t.$$typeof === m || t.$$typeof === s || t.$$typeof === c || t.$$typeof === M || t.$$typeof === k || t.$$typeof === u || t.$$typeof === I);
  }, O.typeOf = R, O;
}
var C = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var he;
function je() {
  return he || (he = 1, process.env.NODE_ENV !== "production" && function() {
    var r = typeof Symbol == "function" && Symbol.for, w = r ? Symbol.for("react.element") : 60103, g = r ? Symbol.for("react.portal") : 60106, a = r ? Symbol.for("react.fragment") : 60107, f = r ? Symbol.for("react.strict_mode") : 60108, i = r ? Symbol.for("react.profiler") : 60114, m = r ? Symbol.for("react.provider") : 60109, s = r ? Symbol.for("react.context") : 60110, E = r ? Symbol.for("react.async_mode") : 60111, p = r ? Symbol.for("react.concurrent_mode") : 60111, c = r ? Symbol.for("react.forward_ref") : 60112, b = r ? Symbol.for("react.suspense") : 60113, A = r ? Symbol.for("react.suspense_list") : 60120, $ = r ? Symbol.for("react.memo") : 60115, j = r ? Symbol.for("react.lazy") : 60116, I = r ? Symbol.for("react.block") : 60121, M = r ? Symbol.for("react.fundamental") : 60117, k = r ? Symbol.for("react.responder") : 60118, u = r ? Symbol.for("react.scope") : 60119;
    function R(n) {
      return typeof n == "string" || typeof n == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      n === a || n === p || n === i || n === f || n === b || n === A || typeof n == "object" && n !== null && (n.$$typeof === j || n.$$typeof === $ || n.$$typeof === m || n.$$typeof === s || n.$$typeof === c || n.$$typeof === M || n.$$typeof === k || n.$$typeof === u || n.$$typeof === I);
    }
    function P(n) {
      if (typeof n == "object" && n !== null) {
        var W = n.$$typeof;
        switch (W) {
          case w:
            var G = n.type;
            switch (G) {
              case E:
              case p:
              case a:
              case i:
              case f:
              case b:
                return G;
              default:
                var pe = G && G.$$typeof;
                switch (pe) {
                  case s:
                  case c:
                  case j:
                  case $:
                  case m:
                    return pe;
                  default:
                    return W;
                }
            }
          case g:
            return W;
        }
      }
    }
    var t = E, L = p, H = s, B = m, V = w, K = c, Q = a, ee = j, re = $, J = g, te = i, F = f, U = b, X = !1;
    function ne(n) {
      return X || (X = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), e(n) || P(n) === E;
    }
    function e(n) {
      return P(n) === p;
    }
    function o(n) {
      return P(n) === s;
    }
    function h(n) {
      return P(n) === m;
    }
    function y(n) {
      return typeof n == "object" && n !== null && n.$$typeof === w;
    }
    function l(n) {
      return P(n) === c;
    }
    function T(n) {
      return P(n) === a;
    }
    function d(n) {
      return P(n) === j;
    }
    function v(n) {
      return P(n) === $;
    }
    function _(n) {
      return P(n) === g;
    }
    function x(n) {
      return P(n) === i;
    }
    function S(n) {
      return P(n) === f;
    }
    function q(n) {
      return P(n) === b;
    }
    C.AsyncMode = t, C.ConcurrentMode = L, C.ContextConsumer = H, C.ContextProvider = B, C.Element = V, C.ForwardRef = K, C.Fragment = Q, C.Lazy = ee, C.Memo = re, C.Portal = J, C.Profiler = te, C.StrictMode = F, C.Suspense = U, C.isAsyncMode = ne, C.isConcurrentMode = e, C.isContextConsumer = o, C.isContextProvider = h, C.isElement = y, C.isForwardRef = l, C.isFragment = T, C.isLazy = d, C.isMemo = v, C.isPortal = _, C.isProfiler = x, C.isStrictMode = S, C.isSuspense = q, C.isValidElementType = R, C.typeOf = P;
  }()), C;
}
var ge;
function Oe() {
  return ge || (ge = 1, process.env.NODE_ENV === "production" ? N.exports = Me() : N.exports = je()), N.exports;
}
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var oe, be;
function ke() {
  if (be)
    return oe;
  be = 1;
  var r = Object.getOwnPropertySymbols, w = Object.prototype.hasOwnProperty, g = Object.prototype.propertyIsEnumerable;
  function a(i) {
    if (i == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(i);
  }
  function f() {
    try {
      if (!Object.assign)
        return !1;
      var i = new String("abc");
      if (i[5] = "de", Object.getOwnPropertyNames(i)[0] === "5")
        return !1;
      for (var m = {}, s = 0; s < 10; s++)
        m["_" + String.fromCharCode(s)] = s;
      var E = Object.getOwnPropertyNames(m).map(function(c) {
        return m[c];
      });
      if (E.join("") !== "0123456789")
        return !1;
      var p = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(c) {
        p[c] = c;
      }), Object.keys(Object.assign({}, p)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return oe = f() ? Object.assign : function(i, m) {
    for (var s, E = a(i), p, c = 1; c < arguments.length; c++) {
      s = Object(arguments[c]);
      for (var b in s)
        w.call(s, b) && (E[b] = s[b]);
      if (r) {
        p = r(s);
        for (var A = 0; A < p.length; A++)
          g.call(s, p[A]) && (E[p[A]] = s[p[A]]);
      }
    }
    return E;
  }, oe;
}
var ie, me;
function le() {
  if (me)
    return ie;
  me = 1;
  var r = "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED";
  return ie = r, ie;
}
var ae, Ee;
function Ce() {
  return Ee || (Ee = 1, ae = Function.call.bind(Object.prototype.hasOwnProperty)), ae;
}
var se, Te;
function qe() {
  if (Te)
    return se;
  Te = 1;
  var r = function() {
  };
  if (process.env.NODE_ENV !== "production") {
    var w = le(), g = {}, a = Ce();
    r = function(i) {
      var m = "Warning: " + i;
      typeof console < "u" && console.error(m);
      try {
        throw new Error(m);
      } catch {
      }
    };
  }
  function f(i, m, s, E, p) {
    if (process.env.NODE_ENV !== "production") {
      for (var c in i)
        if (a(i, c)) {
          var b;
          try {
            if (typeof i[c] != "function") {
              var A = Error(
                (E || "React class") + ": " + s + " type `" + c + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof i[c] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`."
              );
              throw A.name = "Invariant Violation", A;
            }
            b = i[c](m, c, E, s, null, w);
          } catch (j) {
            b = j;
          }
          if (b && !(b instanceof Error) && r(
            (E || "React class") + ": type specification of " + s + " `" + c + "` is invalid; the type checker function must return `null` or an `Error` but returned a " + typeof b + ". You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument)."
          ), b instanceof Error && !(b.message in g)) {
            g[b.message] = !0;
            var $ = p ? p() : "";
            r(
              "Failed " + s + " type: " + b.message + ($ ?? "")
            );
          }
        }
    }
  }
  return f.resetWarningCache = function() {
    process.env.NODE_ENV !== "production" && (g = {});
  }, se = f, se;
}
var ce, _e;
function Ye() {
  if (_e)
    return ce;
  _e = 1;
  var r = Oe(), w = ke(), g = le(), a = Ce(), f = qe(), i = function() {
  };
  process.env.NODE_ENV !== "production" && (i = function(s) {
    var E = "Warning: " + s;
    typeof console < "u" && console.error(E);
    try {
      throw new Error(E);
    } catch {
    }
  });
  function m() {
    return null;
  }
  return ce = function(s, E) {
    var p = typeof Symbol == "function" && Symbol.iterator, c = "@@iterator";
    function b(e) {
      var o = e && (p && e[p] || e[c]);
      if (typeof o == "function")
        return o;
    }
    var A = "<<anonymous>>", $ = {
      array: k("array"),
      bigint: k("bigint"),
      bool: k("boolean"),
      func: k("function"),
      number: k("number"),
      object: k("object"),
      string: k("string"),
      symbol: k("symbol"),
      any: u(),
      arrayOf: R,
      element: P(),
      elementType: t(),
      instanceOf: L,
      node: K(),
      objectOf: B,
      oneOf: H,
      oneOfType: V,
      shape: ee,
      exact: re
    };
    function j(e, o) {
      return e === o ? e !== 0 || 1 / e === 1 / o : e !== e && o !== o;
    }
    function I(e, o) {
      this.message = e, this.data = o && typeof o == "object" ? o : {}, this.stack = "";
    }
    I.prototype = Error.prototype;
    function M(e) {
      if (process.env.NODE_ENV !== "production")
        var o = {}, h = 0;
      function y(T, d, v, _, x, S, q) {
        if (_ = _ || A, S = S || v, q !== g) {
          if (E) {
            var n = new Error(
              "Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types"
            );
            throw n.name = "Invariant Violation", n;
          } else if (process.env.NODE_ENV !== "production" && typeof console < "u") {
            var W = _ + ":" + v;
            !o[W] && // Avoid spamming the console because they are often not actionable except for lib authors
            h < 3 && (i(
              "You are manually calling a React.PropTypes validation function for the `" + S + "` prop on `" + _ + "`. This is deprecated and will throw in the standalone `prop-types` package. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details."
            ), o[W] = !0, h++);
          }
        }
        return d[v] == null ? T ? d[v] === null ? new I("The " + x + " `" + S + "` is marked as required " + ("in `" + _ + "`, but its value is `null`.")) : new I("The " + x + " `" + S + "` is marked as required in " + ("`" + _ + "`, but its value is `undefined`.")) : null : e(d, v, _, x, S);
      }
      var l = y.bind(null, !1);
      return l.isRequired = y.bind(null, !0), l;
    }
    function k(e) {
      function o(h, y, l, T, d, v) {
        var _ = h[y], x = F(_);
        if (x !== e) {
          var S = U(_);
          return new I(
            "Invalid " + T + " `" + d + "` of type " + ("`" + S + "` supplied to `" + l + "`, expected ") + ("`" + e + "`."),
            { expectedType: e }
          );
        }
        return null;
      }
      return M(o);
    }
    function u() {
      return M(m);
    }
    function R(e) {
      function o(h, y, l, T, d) {
        if (typeof e != "function")
          return new I("Property `" + d + "` of component `" + l + "` has invalid PropType notation inside arrayOf.");
        var v = h[y];
        if (!Array.isArray(v)) {
          var _ = F(v);
          return new I("Invalid " + T + " `" + d + "` of type " + ("`" + _ + "` supplied to `" + l + "`, expected an array."));
        }
        for (var x = 0; x < v.length; x++) {
          var S = e(v, x, l, T, d + "[" + x + "]", g);
          if (S instanceof Error)
            return S;
        }
        return null;
      }
      return M(o);
    }
    function P() {
      function e(o, h, y, l, T) {
        var d = o[h];
        if (!s(d)) {
          var v = F(d);
          return new I("Invalid " + l + " `" + T + "` of type " + ("`" + v + "` supplied to `" + y + "`, expected a single ReactElement."));
        }
        return null;
      }
      return M(e);
    }
    function t() {
      function e(o, h, y, l, T) {
        var d = o[h];
        if (!r.isValidElementType(d)) {
          var v = F(d);
          return new I("Invalid " + l + " `" + T + "` of type " + ("`" + v + "` supplied to `" + y + "`, expected a single ReactElement type."));
        }
        return null;
      }
      return M(e);
    }
    function L(e) {
      function o(h, y, l, T, d) {
        if (!(h[y] instanceof e)) {
          var v = e.name || A, _ = ne(h[y]);
          return new I("Invalid " + T + " `" + d + "` of type " + ("`" + _ + "` supplied to `" + l + "`, expected ") + ("instance of `" + v + "`."));
        }
        return null;
      }
      return M(o);
    }
    function H(e) {
      if (!Array.isArray(e))
        return process.env.NODE_ENV !== "production" && (arguments.length > 1 ? i(
          "Invalid arguments supplied to oneOf, expected an array, got " + arguments.length + " arguments. A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z])."
        ) : i("Invalid argument supplied to oneOf, expected an array.")), m;
      function o(h, y, l, T, d) {
        for (var v = h[y], _ = 0; _ < e.length; _++)
          if (j(v, e[_]))
            return null;
        var x = JSON.stringify(e, function(q, n) {
          var W = U(n);
          return W === "symbol" ? String(n) : n;
        });
        return new I("Invalid " + T + " `" + d + "` of value `" + String(v) + "` " + ("supplied to `" + l + "`, expected one of " + x + "."));
      }
      return M(o);
    }
    function B(e) {
      function o(h, y, l, T, d) {
        if (typeof e != "function")
          return new I("Property `" + d + "` of component `" + l + "` has invalid PropType notation inside objectOf.");
        var v = h[y], _ = F(v);
        if (_ !== "object")
          return new I("Invalid " + T + " `" + d + "` of type " + ("`" + _ + "` supplied to `" + l + "`, expected an object."));
        for (var x in v)
          if (a(v, x)) {
            var S = e(v, x, l, T, d + "." + x, g);
            if (S instanceof Error)
              return S;
          }
        return null;
      }
      return M(o);
    }
    function V(e) {
      if (!Array.isArray(e))
        return process.env.NODE_ENV !== "production" && i("Invalid argument supplied to oneOfType, expected an instance of array."), m;
      for (var o = 0; o < e.length; o++) {
        var h = e[o];
        if (typeof h != "function")
          return i(
            "Invalid argument supplied to oneOfType. Expected an array of check functions, but received " + X(h) + " at index " + o + "."
          ), m;
      }
      function y(l, T, d, v, _) {
        for (var x = [], S = 0; S < e.length; S++) {
          var q = e[S], n = q(l, T, d, v, _, g);
          if (n == null)
            return null;
          n.data && a(n.data, "expectedType") && x.push(n.data.expectedType);
        }
        var W = x.length > 0 ? ", expected one of type [" + x.join(", ") + "]" : "";
        return new I("Invalid " + v + " `" + _ + "` supplied to " + ("`" + d + "`" + W + "."));
      }
      return M(y);
    }
    function K() {
      function e(o, h, y, l, T) {
        return J(o[h]) ? null : new I("Invalid " + l + " `" + T + "` supplied to " + ("`" + y + "`, expected a ReactNode."));
      }
      return M(e);
    }
    function Q(e, o, h, y, l) {
      return new I(
        (e || "React class") + ": " + o + " type `" + h + "." + y + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + l + "`."
      );
    }
    function ee(e) {
      function o(h, y, l, T, d) {
        var v = h[y], _ = F(v);
        if (_ !== "object")
          return new I("Invalid " + T + " `" + d + "` of type `" + _ + "` " + ("supplied to `" + l + "`, expected `object`."));
        for (var x in e) {
          var S = e[x];
          if (typeof S != "function")
            return Q(l, T, d, x, U(S));
          var q = S(v, x, l, T, d + "." + x, g);
          if (q)
            return q;
        }
        return null;
      }
      return M(o);
    }
    function re(e) {
      function o(h, y, l, T, d) {
        var v = h[y], _ = F(v);
        if (_ !== "object")
          return new I("Invalid " + T + " `" + d + "` of type `" + _ + "` " + ("supplied to `" + l + "`, expected `object`."));
        var x = w({}, h[y], e);
        for (var S in x) {
          var q = e[S];
          if (a(e, S) && typeof q != "function")
            return Q(l, T, d, S, U(q));
          if (!q)
            return new I(
              "Invalid " + T + " `" + d + "` key `" + S + "` supplied to `" + l + "`.\nBad object: " + JSON.stringify(h[y], null, "  ") + `
Valid keys: ` + JSON.stringify(Object.keys(e), null, "  ")
            );
          var n = q(v, S, l, T, d + "." + S, g);
          if (n)
            return n;
        }
        return null;
      }
      return M(o);
    }
    function J(e) {
      switch (typeof e) {
        case "number":
        case "string":
        case "undefined":
          return !0;
        case "boolean":
          return !e;
        case "object":
          if (Array.isArray(e))
            return e.every(J);
          if (e === null || s(e))
            return !0;
          var o = b(e);
          if (o) {
            var h = o.call(e), y;
            if (o !== e.entries) {
              for (; !(y = h.next()).done; )
                if (!J(y.value))
                  return !1;
            } else
              for (; !(y = h.next()).done; ) {
                var l = y.value;
                if (l && !J(l[1]))
                  return !1;
              }
          } else
            return !1;
          return !0;
        default:
          return !1;
      }
    }
    function te(e, o) {
      return e === "symbol" ? !0 : o ? o["@@toStringTag"] === "Symbol" || typeof Symbol == "function" && o instanceof Symbol : !1;
    }
    function F(e) {
      var o = typeof e;
      return Array.isArray(e) ? "array" : e instanceof RegExp ? "object" : te(o, e) ? "symbol" : o;
    }
    function U(e) {
      if (typeof e > "u" || e === null)
        return "" + e;
      var o = F(e);
      if (o === "object") {
        if (e instanceof Date)
          return "date";
        if (e instanceof RegExp)
          return "regexp";
      }
      return o;
    }
    function X(e) {
      var o = U(e);
      switch (o) {
        case "array":
        case "object":
          return "an " + o;
        case "boolean":
        case "date":
        case "regexp":
          return "a " + o;
        default:
          return o;
      }
    }
    function ne(e) {
      return !e.constructor || !e.constructor.name ? A : e.constructor.name;
    }
    return $.checkPropTypes = f, $.resetWarningCache = f.resetWarningCache, $.PropTypes = $, $;
  }, ce;
}
var ue, Se;
function De() {
  if (Se)
    return ue;
  Se = 1;
  var r = le();
  function w() {
  }
  function g() {
  }
  return g.resetWarningCache = w, ue = function() {
    function a(m, s, E, p, c, b) {
      if (b !== r) {
        var A = new Error(
          "Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types"
        );
        throw A.name = "Invariant Violation", A;
      }
    }
    a.isRequired = a;
    function f() {
      return a;
    }
    var i = {
      array: a,
      bigint: a,
      bool: a,
      func: a,
      number: a,
      object: a,
      string: a,
      symbol: a,
      any: a,
      arrayOf: f,
      element: a,
      elementType: a,
      instanceOf: f,
      node: a,
      objectOf: f,
      oneOf: f,
      oneOfType: f,
      shape: f,
      exact: f,
      checkPropTypes: g,
      resetWarningCache: w
    };
    return i.PropTypes = i, i;
  }, ue;
}
if (process.env.NODE_ENV !== "production") {
  var Le = Oe(), Fe = !0;
  fe.exports = Ye()(Le.isElement, Fe);
} else
  fe.exports = De()();
var We = fe.exports;
const z = /* @__PURE__ */ $e(We);
function Z(r, w) {
  w && (window == null || window._rphConfig.push([r, w]));
}
function Pe({
  appKey: r,
  hubUrlOverride: w,
  stateListener: g,
  locationHash: a,
  ...f
}) {
  return Re(() => {
    if (!window)
      return;
    const i = window._rphConfig = window._rphConfig || [], m = window.localStorage.getItem("rph_base_url_override") || w || "https://hub.rownd.io";
    i.push(["setBaseUrl", m]);
    const s = document, E = s.createElement("script"), p = s.createElement("script"), c = s.getElementsByTagName("script")[0];
    E.noModule = !0, E.async = !0, E.src = m + "/static/scripts/rph.js", p.type = "module", p.async = !0, p.src = m + "/static/scripts/rph.mjs", c != null && c.parentNode ? (c.parentNode.insertBefore(E, c), c.parentNode.insertBefore(p, c)) : (s.body.appendChild(E), s.body.appendChild(p)), Z("setAppKey", r), Z("setStateListener", g), Z("setLocationHash", a), window.localStorage.getItem("rph_log_level") === "debug" && console.debug("[debug] rest:", f), f && (Object.entries(f).forEach(([b, A]) => {
      Z(
        `set${b.charAt(0).toUpperCase() + b.substring(1)}`,
        A
      );
    }), window.localStorage.getItem("rph_log_level") === "debug" && console.debug("[debug] hubConfig:", window._rphConfig));
  }, [r, g, a, w, f]), null;
}
Pe.propTypes = {
  appKey: z.string.isRequired,
  postLoginRedirect: z.string,
  rootOrigin: z.string,
  apiUrl: z.string,
  externalComponent: z.object
};
var we;
const Ue = typeof window < "u" ? (we = window == null ? void 0 : window.location) == null ? void 0 : we.hash : void 0, xe = Ae(void 0);
function ze({ children: r, ...w }) {
  const g = ye(null), a = ye([]), f = Y(
    (u, R) => {
      if (R.length === 1 || u === void 0)
        return u == null ? void 0 : u[R[0]];
      const P = R[0];
      return f(
        u == null ? void 0 : u[P],
        R == null ? void 0 : R.filter((t) => t !== P)
      );
    },
    []
  ), i = Y(
    (u, ...R) => {
      const P = f(g.current, u);
      if (P)
        return P(...R);
      a.current.push({ fnNames: u, args: R });
    },
    [g, f]
  ), m = Y(
    (...u) => i(["requestSignIn"], ...u),
    [i]
  ), s = Y(
    (...u) => i(["getAccessToken"], ...u),
    [i]
  ), E = Y(
    (...u) => i(["signOut"], ...u),
    [i]
  ), p = Y(
    (...u) => i(["user", "manageAccount"], ...u),
    [i]
  ), c = Y(
    (...u) => i(["user", "set"], ...u),
    [i]
  ), b = Y(
    (...u) => i(["user", "setValue"], ...u),
    [i]
  ), A = Y(
    (...u) => i(["firebase", "getIdToken"], ...u),
    [i]
  ), $ = Y(() => i(["getAppConfig"]), [i]), [j, I] = D.useState({
    requestSignIn: m,
    getAccessToken: s,
    signOut: E,
    manageAccount: p,
    setUser: c,
    setUserValue: b,
    getFirebaseIdToken: A,
    getAppConfig: $,
    is_initializing: !0,
    is_authenticated: !1,
    access_token: null,
    auth: {
      access_token: null,
      is_authenticated: !1
    },
    user: {
      data: {},
      groups: [],
      redacted_fields: [],
      verified_data: {},
      meta: {}
    }
  }), M = Y(() => {
    if (a.current.length) {
      for (const { fnNames: u, args: R } of a.current) {
        const P = f(g.current, u);
        if (!P)
          return;
        P(...R);
      }
      a.current.length = 0;
    }
  }, [a, f]), k = Y(
    ({ state: u, api: R }) => {
      var t, L, H, B, V, K;
      const P = {
        // functions
        requestSignIn: m,
        getAccessToken: s,
        signOut: E,
        manageAccount: p,
        setUser: c,
        setUserValue: b,
        getFirebaseIdToken: A,
        getAppConfig: $,
        // data
        is_initializing: u.is_initializing,
        is_authenticated: !!((t = u.auth) != null && t.access_token),
        access_token: ((L = u.auth) == null ? void 0 : L.access_token) || null,
        auth: {
          access_token: (H = u.auth) == null ? void 0 : H.access_token,
          app_id: (B = u.auth) == null ? void 0 : B.app_id,
          is_authenticated: !!((V = u.auth) != null && V.access_token),
          is_verified_user: (K = u.auth) == null ? void 0 : K.is_verified_user
        },
        user: {
          ...u.user
        }
      };
      I(P), g.current = R, M();
    },
    [
      M,
      s,
      m,
      E,
      A,
      p,
      c,
      b
    ]
  );
  return (window == null ? void 0 : window.localStorage.getItem("rph_log_level")) === "debug" && console.debug("[debug] rph_txstate:", j), /* @__PURE__ */ D.createElement(xe.Provider, { value: j }, /* @__PURE__ */ D.createElement(
    Pe,
    {
      stateListener: k,
      locationHash: Ue,
      ...w
    }
  ), r);
}
ze.propTypes = {
  appKey: z.string.isRequired,
  children: z.node.isRequired
};
function de() {
  const r = Ie(xe);
  if (r === void 0)
    throw new Error("useRownd must be used within a RowndProvider");
  return r;
}
const Be = ({
  children: r,
  initializing: w,
  signInProps: g
}) => {
  const { is_authenticated: a, is_initializing: f, requestSignIn: i } = de();
  return Re(() => {
    !a && !f && i({ prevent_closing: !0, ...g });
  }, [a, f, i, g]), f && w ? /* @__PURE__ */ D.createElement(D.Fragment, null, w) : /* @__PURE__ */ D.createElement(D.Fragment, null, r);
}, Ve = ({ children: r }) => {
  const { is_authenticated: w } = de();
  return w ? /* @__PURE__ */ D.createElement(D.Fragment, null, r) : null;
}, Ke = ({ children: r }) => {
  const { is_authenticated: w } = de();
  return w ? null : /* @__PURE__ */ D.createElement(D.Fragment, null, r);
};
export {
  Be as RequireSignIn,
  ze as RowndProvider,
  Ve as SignedIn,
  Ke as SignedOut,
  de as useRownd
};
